### PR TITLE
add consul integration docs

### PR DESCRIPTION
**Pull request by sudhanshu456**
### Summary by CodeRabbit

- New Feature: Added a comprehensive guide for Consul integration. This includes detailed instructions on prerequisites, installation steps, verification of the installation, and uninstallation process.
- Documentation: Provided support information for different versions and service defaults. Also included are visual aids with images for both light and dark modes to enhance user understanding.

### CODE REVIEW

**Review by tanveergill — APPROVED** [review]

File: docs/content/get-started/installation/supported-platforms.md [hunk 1]
```diff
@@ -14,7 +14,6 @@ Below are the platforms supported for the Aperture Agent:
 | Platform   | Extent of Support   |
 | ---------- | ------------------- |
 | Kubernetes | 1.23 and above      |
-| Istio      | 1.14 and above      |
```
Comment (sudhanshu456) [inline docs/content/get-started/installation/supported-platforms.md:17]: any reason why we removed it? @hdkshingala

Comment (hdkshingala) [reply docs/content/get-started/installation/supported-platforms.md:17]: That support-platform was just for installation. We have version for istio under it's own page

File: docs/content/integrations/consul/consul.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+   spec:
+     protocol: grpc
+   ```
+
+2. The below Service Defaults adds
+   [OpenTelemetry Access Logging](https://developer.hashicorp.com/consul/docs/v1.17.x/connect/proxies/envoy-extensions/usage/otel-access-logging)
+   Envoy Extension for the outbound listener, in the Consul Proxy running with
+   the application.
+
+   The Open Telemetry configuration in the following has extracted values, which
```
Comment (sudhanshu456) [inline docs/content/integrations/consul/consul.md:82]:
```suggestion
   The OpenTelemetry configuration in the following has extracted values, which
```

File: docs/content/integrations/consul/consul.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+             name: aperture-agent
+             namespace: aperture-agent
+   ```
+
+3. The below Service Defaults adds
+   [OpenTelemetry Access Logging](https://developer.hashicorp.com/consul/docs/v1.17.x/connect/proxies/envoy-extensions/usage/otel-access-logging)
+   Envoy Extension, but for the inbound listener, in the Consul Proxy running
+   with the application.
+
+   The Open Telemetry configuration in the following has extracted values, which
```
Comment (sudhanshu456) [inline docs/content/integrations/consul/consul.md:121]:
```suggestion
   The OpenTelemetry configuration in the following has extracted values, which
```

File: docs/content/integrations/consul/consul.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+             name: aperture-agent
+             namespace: aperture-agent
+         initialMetadata:
+           - key: control-point
+             value: "ingress"
+   ```
+
+## Installation {#installation}
+
+The complete Service Defaults configuration for the Consul Proxy running with
```
Comment (sudhanshu456) [inline docs/content/integrations/consul/consul.md:217]: We should mention it is example configuration.

File: pkg/policies/flowcontrol/service/envoy/authz.go [hunk 5]
```diff
@@ -160,6 +160,14 @@ func (h *Handler) Check(ctx context.Context, req *authv3.CheckRequest) (*authv3.
 	flowlabel.Merge(mergedFlowLabels, sdFlowLabels)
 
 	svcs := h.serviceGetter.ServicesFromContext(ctx)
+	if len(svcs) == 0 {
```
[the remark this answers is not in the document]
Comment (hdkshingala) [reply pkg/policies/flowcontrol/service/envoy/authz.go:163]: Yes that's why it is in a separate commit and a comment with reason

**Pull request merged**